docs/source/llms/dspy/notebooks/dspy_quickstart.ipynb

```diff
@@ -0,0 +1,1265 @@
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "fa602d14-6372-4e0a-92d8-6f7db048c3b8",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "[DSPy](https://dspy-docs.vercel.app/) simplifies building language model (LM) pipelines by replacing manual prompt engineering with structured \"text transformation graphs.\" These graphs use flexible, learning modules that automate and optimize LM tasks like reasoning, retrieval, and answering complex questions. \n",
+    "\n",
+    "### How does it work?\n",
+    "At a high level, DSPy optimizes prompts, selects the best language model, and can even fine-tune the model using training data.\n",
+    "\n",
+    "The process follows these three steps, common to most DSPy [optimizers](https://dspy-docs.vercel.app/docs/building-blocks/optimizers):\n",
+    "\n",
+    "1. **Candidate Generation**: DSPy finds all `Predict` modules in the program and generates variations of instructions and demonstrations (e.g., examples for prompts). This step creates a set of possible candidates for the next stage.\n",
+    "2. **Parameter Optimization**: DSPy then uses methods like random search, TPE, or Optuna to select the best configurations. Fine-tuning models can also be done at this stage.\n",
+    "3. **Higher-Order Optimization**: Lastly, DSPy can modify the program's structure, such as creating ensembles of different variations of the program to boost performance.\n",
+    "\n",
+    "### This Demo\n",
+    "Below we create a simple program that demonstrates the power of DSPy. We will build a text classifier leveraging OpenAI. By the end of this tutorial, we will...\n",
+    "\n",
+    "1. Define a [dspy.Signature](https://dspy-docs.vercel.app/docs/building-blocks/signatures) and [dspy.Module](https://dspy-docs.vercel.app/docs/building-blocks/modules) to perform text classification.\n",
+    "2. Leverage [dspy.teleprompt.BootstrapFewShotWithRandomSearch](https://dspy-docs.vercel.app/api/optimizers/BootstrapFewShotWithRandomSearch) to compile our module so it's better at classifying our text.\n",
+    "3. Log the compiled model with MLflow.\n",
+    "4. Load the compiled model with MLflow and perform inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "58918566-6e93-4a2e-9a34-b0f56378885a",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "zsh:1: 0.2.5 not found\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%pip install -U openai dspy-ai>=0.2.5 mlflow>=2.17.0"
```

Comment:
let's use `dspy`, since 2.5.1 `dspy` becomes the golden package name instead of `dspy-ai`